addons/test_mail/tests/test_mail_gateway.py

```diff
@@ -81,6 +81,19 @@ def test_message_parse_body(self):
         res = self.env['mail.thread'].message_parse(self.from_string(test_mail_data.MAIL_MULTIPART_WEIRD_FILENAME))
         self.assertEqual(res['attachments'][0][0], '62_@;,][)=.(ÇÀÉ.txt')
 
+    def test_pdf_mime_type_aliasing(self):
+        # This test checks aliasing content-type (mime type) of "applicaiton/pdf" with just "pdf" works correctly.
+
+        # Baseline check. Parsing mail with "applicaiton/pdf"
+        mail_with_standard_mime = self.format(test_mail_data.MAIL_PDF_MIME_TEMPLATE, pdf_mime="application/pdf")
+        res_std = self.env['mail.thread'].message_parse(self.from_string(mail_with_standard_mime))
+        self.assertEqual(res_std['attachments'][0].content, test_mail_data.PDF_PARSED, "Attachment with Content-Type: application/pdf parsed with error")
```

Comment:
I've seen people write positive statements for the 3rd (message) argument of assertEqual "Attachment parsed correctly", I've also seen other people use write a negative statement "Attachment parsed with error" and nowadays I find both ways confusing (due to the lack of coherence).

May I suggest the following instead:
```suggestion
        self.assertEqual(res_std['attachments'][0].content, test_mail_data.PDF_PARSED, "Attachment with Content-Type: application/pdf must parse without error")
```

If you have it inside `res_std`, can you also verify that the computed mimetype is indeed `application/pdf` and not `text/plain`?

Reply:
> If you have it inside res_std, can you also verify that the computed mimetype is indeed application/pdf and not text/plain?
About this one. So to answer the main question the mimetype is **not** there. 

It is possible to add it in there, for example like that:
![image](https://github.com/odoo/odoo/assets/33809926/4ead4ac0-a426-4701-a597-85fecd58cdbd)

Is it worth adding it, just for this check? I don't know what to think about it.

code from the picture
https://github.com/odoo/odoo/blob/d30682b41e51e77f3c28e30bae558b0dd8257d2b/addons/mail/models/mail_thread.py#L1286-L1295

Reply:
Side note on error msg: if we add one, it would give a hint on what failed (aka here the fallback parsing allowing to find pdf instead of text/plain). Otherwise if it raises, we obviously know there is an error ^^ . Msg should add information allowing to understand the context / test / ...

Reply:
> Is it worth adding it, just for this check? I don't know what to think about it.

No, let's keep it this way